units/src/amount/signed.rs

```diff
@@ -158,7 +158,7 @@ impl SignedAmount {
     pub fn from_str_in(s: &str, denom: Denomination) -> Result<SignedAmount, ParseAmountError> {
         match parse_signed_to_satoshi(s, denom).map_err(|error| error.convert(true))? {
             // (negative, amount)
-            (false, sat) if sat > SignedAmount::MAX.to_sat() as u64 => Err(ParseAmountError(
+            (false, sat) if sat > SignedAmount::MAX_MONEY.to_sat() as u64 => Err(ParseAmountError(
```

Comment:
`::MAX` looks a lot like `u64::MAX` and for Bitcoin devs an amount is intrinsically a 64 byte unsigned integer.

Maybe, but not for Rust devs, where `MAX` represents maximum value a *type* can hold. I find it an argument **for** `MAX`.

The compiler will prevent any mixups. But we should deprecate anything in older versions that unconditionally produces amounts > 21M BTC.

Reply:
That one doesn't produce high amounts **unconditionally**.

Reply:
So 2 votes for using `MAX` and not `MAX_MONEY`. @apoelstra do you want to weigh in, if not I'll change it to that.

Reply:
BTW in any case I think we'll need to deprecate `Amount::MAX` in the old version saying it's creating absurdly large amounts and its semantics will change.

Reply:
Only use `MAX` and at the same time deprecate `MAX` - you've lost me?

Reply:
> Only use MAX and at the same time deprecate MAX - you've lost me?

I think he's pointing out that if you're using Amount::MAX for values greater than MAX_MONEY now, and the the API changes to limit it to Amount::MAX to MAX_MONEY that could be a surprise (since this const is public).

Reply:
Now uses MAX everywhere and references this thread in the commit log to justify the change.

Reply:
@yancyribbens exactly!

Reply:
I think there is the possibility for catastrophe changing MAX from u64::MAX to MAX_MONEY without deprecation.  Someone could be using this constant thinking that what they are getting is u64::MAX doing some computations etc.  After upgrading they are going to get a completely different result without any warning.  However, that change to `MAX` was done a while ago: https://github.com/rust-bitcoin/rust-bitcoin/pull/3719